File: Java/QuizGame/src/partite/SinglePlayer.java
package partite;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

import gameReport.ReportSP;
import utils.ConfigReader;
import utils.ConsoleCleaner;
import utils.QuestionReader;

public class SinglePlayer extends Partita{
	public SinglePlayer(String path) {
		this.quizPath = path;
		ConfigReader cr = new ConfigReader(path + "\\config.txt");
		this.nDomande = cr.getNDomande();
	}
	
	@Override
	public ReportSP start(Scanner in) {
		QuestionReader qr = new QuestionReader(quizPath);
		
		String[] reportStrings = new String[nDomande];
		String rispGiusta;
		String domanda;
		int nRispGiuste = 0;
		
		for(int i=0;i<nDomande;i++) {
			ArrayList<String> q = qr.getDomanda(i+1);
			rispGiusta = q.get(1);
			domanda = q.get(0);
			
			q.remove(0);
			Collections.shuffle(q);
						
			int rispData;
			do {
				System.out.println("Domanda "+(i+1)+": "+domanda);
				System.out.println("1) "+q.get(0)+"\t\t\t2)"+q.get(1));
				System.out.println("3) "+q.get(2)+"\t\t\t4)"+q.get(3));
				System.out.print("Inserisci la risposta: (1-2-3-4): ");
				while (!in.hasNextInt()) {
					ConsoleCleaner.clear();
					System.out.println("Domanda "+(i+1)+": "+domanda);
					System.out.println("1) "+q.get(0)+"\t\t\t2)"+q.get(1));
					System.out.println("3) "+q.get(2)+"\t\t\t4)"+q.get(3));
					System.out.print("Inserisci la risposta: (1-2-3-4): ");
					in.next();
				}
				rispData = in.nextInt();
				ConsoleCleaner.clear();
			}while(rispData<1 || rispData>4);
			
	
			reportStrings[i] = buildReportString(i+1,domanda,q.get(rispData-1),rispGiusta);
			
			if(q.get(rispData-1).equals(rispGiusta)) {
				nRispGiuste++;
			}
			
		}
		ReportSP report = new ReportSP(nRispGiuste, reportStrings);
		return report;
	}

	private String buildReportString(int nDomanda,String domanda, String rispostaData, String rispostaGiusta) {
		String res = nDomanda + ") "+ domanda + " " + rispostaData + ". Risposta giusta: "+rispostaGiusta;
		return res;
	}

}

File: C++/QuizGame/src/partite/SinglePlayer.cpp
#include "SinglePlayer.h"
#include "../utils/ConfigReader.h"
#include "../utils/QuestionReader.h"
#include <iostream>
#include <vector>
#include <algorithm>

SinglePlayer::SinglePlayer(std::string path) {
	this->path = path;
	ConfigReader cr(path);
	this->nDomande = cr.getNDomande();
}

void SinglePlayer::start(){
	int nRispGiuste = 0;

	for(int i=0;i<nDomande;i++){
		if(showDomanda(i)){
			nRispGiuste+=1;
		}
	}
	std::cin.ignore();
	std::cout<<"Risultato: "<<nRispGiuste<<"/"<<nDomande<<std::endl<<"Premi invio per continuare...\n"<<std::flush;
	std::cin.get();
	system("CLS");
}

bool SinglePlayer::showDomanda(int i){

	QuestionReader qr(path);

	std::string rispGiusta;
	std::string domanda;
	int rispData;

	std::vector<std::string> q = qr.getDomanda(i+1);
	rispGiusta = q[1];
	domanda = q[0];

	q.erase(q.begin());
	std::random_shuffle(q.begin(), q.end());

	std::cout<<"Domanda "<<i+1<<")"<<domanda<<"\n"
			"1) "<<q[0]<<"\t\t\t2) "<<q[1]<<"\n"
			"3) "<<q[2]<<"\t\t\t4) "<<q[3]<<"\n\n";
	std::cin>>rispData;
	while(!rispData || (rispData<1||rispData>4)){
		system("CLS");
		std::cout<<"Domanda "<<i+1<<")"<<domanda<<"\n"
				"1) "<<q[0]<<"\t\t\t2) "<<q[1]<<"\n"
				"3) "<<q[2]<<"\t\t\t4) "<<q[3]<<"\n\n";
		std::cin.clear();
		std::cin.ignore(40,'\n');
		std::cin>>rispData;
	}
	system("CLS");

	if(q[rispData-1] == rispGiusta){
		return true;
	}
	return false;

}

SinglePlayer::~SinglePlayer() {
}

